app/scripts/controllers/app-state-controller.ts

```diff
@@ -0,0 +1,860 @@
+  // multiple networks.
+  hadAdvancedGasFeesSetPriorToMigration92_3: boolean;
+  qrHardware: Json;
+  nftsDropdownState: Json;
+  usedNetworks: Record<string, boolean>;
+  surveyLinkLastClickedOrClosed: number | null;
+  signatureSecurityAlertResponses: Record<string, SecurityAlertResponse>;
+  // States used for displaying the changed network toast
+  switchedNetworkDetails: Record<string, string> | null;
+  switchedNetworkNeverShowMessage: boolean;
+  currentExtensionPopupId: number;
+  lastInteractedConfirmationInfo?: LastInteractedConfirmationInfo;
+  termsOfUseLastAgreed?: number;
+  snapsInstallPrivacyWarningShown?: boolean;
+  interactiveReplacementToken?: { url: string; oldRefreshToken: string };
+  noteToTraderMessage?: string;
+  custodianDeepLink?: { fromAddress: string; custodyId: string };
+};
+
+const controllerName = 'AppStateController';
+
+/**
+ * Returns the state of the {@link AppStateController}.
+ */
+export type AppStateControllerGetStateAction = {
+  type: 'AppStateController:getState';
+  handler: () => AppStateControllerState;
+};
+
+/**
+ * Actions exposed by the {@link AppStateController}.
+ */
+export type AppStateControllerActions = AppStateControllerGetStateAction;
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions = AddApprovalRequest | AcceptRequest;
+
+/**
+ * Event emitted when the state of the {@link AppStateController} changes.
+ */
+export type AppStateControllerStateChangeEvent = {
+  type: 'AppStateController:stateChange';
+  payload: [AppStateControllerState, []];
+};
+
+/**
+ * Events emitted by {@link AppStateController}.
+ */
+export type AppStateControllerEvents = AppStateControllerStateChangeEvent;
+
+/**
+ * Events that this controller is allowed to subscribe.
+ */
+export type AllowedEvents =
+  | PreferencesControllerStateChangeEvent
+  | KeyringControllerQRKeyringStateChangeEvent;
+
+export type AppStateControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AppStateControllerActions | AllowedActions,
+  AppStateControllerEvents | AllowedEvents,
+  AllowedActions['type'],
+  AllowedEvents['type']
+>;
+
+type PollingTokenType =
+  | 'popupGasPollTokens'
+  | 'notificationGasPollTokens'
+  | 'fullScreenGasPollTokens';
+
+type SecurityAlertResponse = {
+  block?: number;
+  description?: string;
+  features?: string[];
+  providerRequestsCount?: Record<string, number>;
+  reason: string;
+  result_type: string;
+  securityAlertId?: string;
+  source?: SecurityAlertSource;
+};
+
+type AppStateControllerOptions = {
+  addUnlockListener: (callback: () => void) => void;
+  isUnlocked: () => boolean;
+  initState?: Partial<AppStateControllerState>;
+  onInactiveTimeout?: () => void;
+  // TODO: Remove this as soon as PreferencesController upgrade to BaseControllerV2 merges with develop
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  preferencesStore: any;
```

Comment:
I believe this can have the `ObservableStore<PreferencesControllerState>` type, regardless of the BaseControllerV2 upgrade